includes/wc-term-functions.php

```diff
@@ -689,3 +690,40 @@ function _wc_recount_terms_by_product( $product_id = '' ) {
 		_wc_term_recount( $product_tags, get_taxonomy( 'product_tag' ), false, false );
 	}
 }
+
+/**
+ * Assigns default product category for products
+ * that have no categories.
+ *
+ * @since 5.4
+ * @return void
+ */
+function _wc_maybe_assign_default_product_cat() {
```

Comment:
Instead of creating a new "private" (as indicated by the initial underscore in the name) function, it would be cleaner and easier to maintain/refactor to create a new class in the `src/Internal` directory. The action scheduling part can be left as is.

Reply:
So I am just following consistency here as seen in https://github.com/woocommerce/woocommerce/blob/trunk/includes/wc-term-functions.php#L392 and https://github.com/woocommerce/woocommerce/blob/trunk/includes/wc-term-functions.php#L152

Reply:
Ok I have converted the PR into using the internal DI container.